setup.py

```diff
@@ -16,17 +16,19 @@
 from setuptools import find_packages
 from setuptools import setup
 
+exec(open('opencensus/common/version/__init__.py').read())
+
 extras = {
-    "stackdriver": ['google-cloud-trace>=0.20.1, <0.30'],
-    "prometheus_client": ['prometheus_client==0.3.1'],
-    "requests": ['wrapt==1.10.11']
+    'prometheus_client': ['prometheus_client==0.3.1'],
+    'requests': ['wrapt==1.10.11'],
+    'stackdriver': ['google-cloud-trace>=0.20.1, <0.30'],
 }
 
 install_requires = [
+    'opencensus-common-correlationcontext == {}'.format(__version__),
```

Comment:
I'm surprised you want to keep the versions in sync.

So installing `opencensus` will install `opencensus-common-correlationcontext`, but not vice-versa. Other packages like exporters will install `opencensus` on install, but not vice-versa. Is that right?

I think you could shorten the distribution name to e.g. `opencensus-correlation`, or is there a reason you need to match the package name? In any case that's a matter of style, up to you.

Reply:
I actually plan to extract the version to a common place and share it across all packages in this repo (with the cost of generating the version.py file on the fly during build/packaging/publishing). Packages in different repo (e.g. opencensus-ext-microsoft-azure-exporter) can have their own version. Do you think this is a good or bad idea?

Yes, `opencensus` depends on `opencensus-common-correlationcontext`. Exporters will depend on `opencensus`. Currently I'm doing dirty job in nox.py in order to get local installation/test working. I haven't yet found a solution to tell pip to look for a specified local path before going to the PyPI indexer.

The pros: all packages will have the same version, easy to maintain and troubleshoot.
The cons: even if we don't update the code for certain package, the version will still increase on each release.

Reply:
As you have observed, I'm trying to match the package name with the namespace - e.g. namespace foo.bar.baz will have package name foo-bar-baz. I thought this would help other plugin/extension developers to keep names consistent.

Do you have a better idea? I've always had problem naming things.

Reply:
I think one of the big benefits of having a different distribution for each package is that they can have different version numbers, especially because of the semantics for breaking changes. If e.g. we release a major- or minor-version change to `foo` and a patch-version update to `bar` users should still be able to safely update `bar` without changing application code. They'd have to go look at the changelog to tell this if we keep the version numbers in sync.

I think it's fine to have different names for the distribution and package, and personally prefer short names.

Reply:
Per our discussion:
1. Changed to short package name `opencensus-correlation`.
2. Each package will have its independent version and changelog. (This would benefit us in the long run if we have git submodule).